src/test/java/net/minecraftforge/debug/entity/living/VillagerChangeProfessionEventTest.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.debug.entity.living;
+
+import com.mojang.logging.LogUtils;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.event.entity.living.VillagerEvent;
+import net.minecraftforge.fml.common.Mod;
+import org.slf4j.Logger;
+
+@Mod("villager_change_profession_event_test")
+public class VillagerChangeProfessionEventTest {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    public VillagerChangeProfessionEventTest() {
+        MinecraftForge.EVENT_BUS.addListener(this::onEvent);
+    }
+
+    public void onEvent(VillagerEvent.ChangeProfessionEvent event) {
+        LOGGER.info("Changed profession from {} to {}", event.getOldProfession(), event.getNewProfession());
```

Comment:
Could this test also modify the profession to make sure you actually can modify it?